API_OVERVIEW.md

```diff
@@ -0,0 +1,175 @@
+# Envoy v2 JSON REST and gRPC APIs
```

Comment:
I think that these docs and the overview serve two distinct purposes. One is conveying to folks who want to understand history and design of the API, who may want to  contribute changes and need this context, what the state of the world is. The other is a user guide with a much more condensed TL;DR. This refactor/move is fine, but I don't think we're  going to be talking about principles and unimplemented N^2 health check scaling in the v2 overview user docs.

Reply:
I agree. I don't think this file goes away, though I do think it gets cleaned up as the official docs come online.